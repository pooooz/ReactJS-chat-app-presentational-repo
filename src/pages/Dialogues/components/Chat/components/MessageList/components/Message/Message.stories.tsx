import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Message } from './Message';

export default {
  title: 'pages/Dialogues/components/Message',
  component: Message,
} as ComponentMeta<typeof Message>;

const message = { id: '1', author: 'Admin', text: 'Initialization...' };
export const Default: ComponentStory<typeof Message> = () => (
  <Message message={message} />
);
