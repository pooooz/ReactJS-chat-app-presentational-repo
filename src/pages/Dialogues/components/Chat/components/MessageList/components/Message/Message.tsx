import React, { FC } from 'react';

import styles from './Message.module.scss';

export interface Message {
  id: string;
  author: string;
  text: string;
}

interface MessageProps {
  message: Message;
}

export const Message: FC<MessageProps> = (props) => {
  return (
    <div className={styles.message}>
      <p className={styles.glitch} data-text={props.message.text}>
        {props.message.text}
      </p>
      <p
        className={styles.glitch}
        data-text={'Author: <' + props.message.author + '>'}
      >
        Author: &lt;{props.message.author}&gt;
      </p>
    </div>
  );
};

export const MessagePure = React.memo(Message);
