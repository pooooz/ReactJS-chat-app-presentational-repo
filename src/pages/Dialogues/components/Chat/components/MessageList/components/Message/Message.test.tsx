import '@testing-library/jest-dom';
import React from 'react';
import { render, screen } from '@testing-library/react';
import { Message } from './Message';

const message = {
  id: '1',
  author: 'Test',
  text: 'Test',
};

describe('Message', () => {
  it('Render component', () => {
    render(<Message message={message} />);
  });

  it('Render with snapshot', () => {
    const { asFragment } = render(<Message message={message} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Render component with author', () => {
    render(<Message message={message} />);
    expect(screen.getByText(/Author: <\w*>/)).toBeInTheDocument();
  });
});
