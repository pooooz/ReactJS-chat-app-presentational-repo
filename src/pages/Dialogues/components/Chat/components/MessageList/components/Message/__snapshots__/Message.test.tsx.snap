// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Render with snapshot 1`] = `
<DocumentFragment>
  <div
    class="message"
  >
    <p
      class="glitch"
      data-text="Test"
    >
      Test
    </p>
    <p
      class="glitch"
      data-text="Author: <Test>"
    >
      Author: &lt;Test&gt;
    </p>
  </div>
</DocumentFragment>
`;
